### PR TITLE
Update to latest dcrd modules

### PR DESCRIPTION
**Pull request by jrick**

### CODE REVIEW

File: wallet/wallet.go [hunk 1]
```diff
@@ -3759,7 +3759,12 @@ func isVote(tx *wire.MsgTx) bool {
 }
 
 func isRevocation(tx *wire.MsgTx) bool {
-	return stake.IsSSRtx(tx)
+	// isAutoRevocationsEnabled is set false to keep manually generated
+	// revocations still considered to be valid revocations.  Enabling this
+	// flag would cause legacy revocations to not be flagged as such.
+	// Enabling the flag only adds additional checks, so it's not necessary
+	// to make the call twice with the flag enabled and disabled.
```
Comment (jrick) [inline wallet/wallet.go:3766]: this comment applies to all of the changes throughout the wallet codebase as well where this flag is now necessary.

File: go.mod [hunk 2]
```diff
@@ -4,25 +4,26 @@ go 1.16
...
+	github.com/decred/dcrd/blockchain/v4 v4.0.0-20210914212651-723d86274b0d
+	github.com/decred/dcrd/certgen v1.1.2-0.20210914212651-723d86274b0d
+	github.com/decred/dcrd/chaincfg/chainhash v1.0.4-0.20210914212651-723d86274b0d
+	github.com/decred/dcrd/chaincfg/v3 v3.0.1-0.20210914212651-723d86274b0d
+	github.com/decred/dcrd/connmgr/v3 v3.0.1-0.20210914212651-723d86274b0d
+	github.com/decred/dcrd/crypto/blake256 v1.0.1-0.20210914212651-723d86274b0d
+	github.com/decred/dcrd/crypto/ripemd160 v1.0.2-0.20210914212651-723d86274b0d
+	github.com/decred/dcrd/database/v3 v3.0.0-20210914212651-723d86274b0d // indirect
+	github.com/decred/dcrd/dcrec v1.0.1-0.20210914212651-723d86274b0d
+	github.com/decred/dcrd/dcrec/secp256k1/v4 v4.0.1-0.20210914212651-723d86274b0d
```
Comment (chappjc) [inline go.mod:19]: There are no changes toward a secp256k1 v4.0.1 yet, so I suspect you want to require that actual v4.0.0 tag, just like with dcrjson/v4

**Review by chappjc — APPROVED** [review]
LGTM.  We used part of that reasoning for isAutoRevocationsEnabled=false in dcrdex too, or will in https://github.com/decred/dcrdex/pull/1211

File: go.mod (hunk 2 above)
Comment (jrick) [reply go.mod:19]: i have a script that just updates everything to the latest master. once we prepare for the release, all of these will be switched to using released versions.

**Pull request merged**